extensions/config-viewer/runtime/src/main/java/io/quarkus/config/ConfigViewerServlet.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.quarkus.config;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.json.Json;
+import javax.json.JsonBuilderFactory;
+import javax.json.JsonObject;
+import javax.json.JsonWriter;
+import javax.json.JsonWriterFactory;
+import javax.json.stream.JsonGenerator;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.microprofile.config.Config;
+
+/**
+ * Servlet, which returns all config sources and properties as JSON. The config sources are sorted descending by ordinal,
+ * the properties by name. If no config is defined an empty JSON object is returned.
+ *
+ * <p>
+ * A typical output might look like:
+ * </p>
+ *
+ * <pre>
+ * {
+ *   "sources": [
+ *     {
+ *       "source": "SysPropConfigSource",
+ *       "ordinal": 400,
+ *       "properties": {
+ *         "file.encoding": "UTF-8",
+ *         "file.separator": "/"
+ *       }
+ *     },
+ *     {
+ *       "source": "EnvConfigSource",
+ *       "ordinal": 300,
+ *       "properties": {
+ *         "EDITOR": "vim",
+ *         "LC_ALL": "en_US.UTF-8"
+ *       }
+ *     },
+ *     {
+ *       "source": "PropertiesConfigSource[source=resource:META-INF/microprofile-config.properties]",
+ *       "ordinal": 100,
+ *       "properties": {
+ *         "greeting.message": "hello",
+ *         "greeting.name": "quarkus"
+ *       }
+ *     }
+ *   ]
+ * }
+ * </pre>
+ */
+@WebServlet
```

Comment:
Ideally I think this should be a pure Vert.x Handler like Health and Metrics: https://github.com/quarkusio/quarkus/blob/master/extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/SmallRyeHealthHandler.java

And thus change the dependency from undertow to vertx. That's going to be the lowest common denominator

Reply:
I'm using a pure Vert.x handler now.